src/registration.ts

```diff
@@ -153,7 +159,12 @@ export class Registration<Shop extends ShopInterface = ShopInterface> {
 	public async delete(req: Request): Promise<Response> {
 		const ctx = await this.app.contextResolver.fromAPI(req);
 
-		await this.app.repository.deleteShop(ctx.shop.getShopId());
+		const event = new AppUninstallEvent(req, ctx.shop);
+		await this.app.hooks.publish("onAppUninstall", event);
+
+		if (!event.keepShop) {
```

Comment:
We don't get the `keepUserData` property as part of the `app.deleted` webhook, unfortunately.

In my use case I have no user data, though, but I would like to keep the shop in my storage, because I have additional state stored alongside it that I want to keep, e.g. in case the app is installed again.

@shyim WDYT?